### PR TITLE
Ordinals v2

### PR DESCRIPTION
**Pull request by kobby-pentangeli**
# Motivation

Closes #

## Changes

- [x] Directory structure and nomenclature
- [x] Implementation of `NFT`-type inscriptions
- [x] New transaction builder
- [x] New transaction signer

## TODO

- [x] Add more unit tests
- [x] Implement the `Indexer`

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Review by veeso — CHANGES_REQUESTED** [review]

File: src/inscription/nft.rs [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+    /// Has a tag of 2, representing the position of the inscribed sat in the outputs.
+    pub pointer: Option<Vec<u8>>,
+    /// Has a tag of 3, representing the parent NFT, i.e., the owner of an NFT
+    /// can create child NFT.
+    pub parent: Option<Vec<u8>>,
+    /// Has a tag of 5, representing CBOR metadata, stored as data pushes.
+    pub metadata: Option<Vec<u8>>,
+    /// Has a tag of 7, representing the metaprotocol identifier.
+    pub metaprotocol: Option<Vec<u8>>,
+    pub incomplete_field: bool,
```
[the remark this answers is not in the document]
Comment (kobby-pentangeli) [reply src/inscription/nft.rs:44]: Yes, linked in the `References` section of the `README`. I'll add the docs to the `_field` fields now...

File: examples/deploy.rs [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-use ord_rs::OrdTransactionBuilder;
+use crate::utils::{calc_fees, Fees};
 use utils::rpc_client;
 
-use crate::utils::{calc_fees, Fees};
+use argh::FromArgs;
+use bitcoin::{secp256k1::Secp256k1, Address, Network, PrivateKey};
+use log::{debug, info};
+use ord_rs::{
+    wallet::{CreateCommitTransactionArgs, RevealTransactionArgs},
```
[the remark this answers is not in the document]
Comment (kobby-pentangeli) [reply examples/deploy.rs:10]: Done

File: src/inscription/nft.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kobby-pentangeli) [reply src/inscription/nft.rs:54]: Done

File: rustfmt.toml [hunk 3]
```diff
@@ -1,2 +1,9 @@
-imports_granularity = "Module"
```
Comment (veeso) [inline rustfmt.toml:1]: those two lines are the configuration we use in our projects

**Review by veeso — CHANGES_REQUESTED** [review]

File: rustfmt.toml (hunk 3 above)
Comment (kobby-pentangeli) [reply rustfmt.toml]: Yeah I know. Previously (before I added the nightly profile in `rust-toolchain.toml`), Cargo complained that those were unstable language features.

Nevertheless, I've reverted to the old. Running `cargo fmt` now works as expected!

**Review by veeso — APPROVED** [review]

**Review by ufoscout — APPROVED** [review]

**Pull request merged**